DataPlotly/data_plotly_dialog.py

```diff
@@ -984,8 +1031,85 @@ def get_settings(self) -> PlotSettings:
             'bargaps': self.bar_gap.value()
         }
 
+        layout_properties['show_legend_check'] = self.show_legend_check.isChecked()
+        layout_properties['orientation_legend_check'] = self.orientation_legend_check.isChecked()
+        layout_properties['range_slider_combo'] = self.range_slider_combo.isChecked()
+        layout_properties['info_combo'] = self.info_combo.currentText()
+        layout_properties['additional_info_combo'] = self.additional_info_combo.expression()
+        layout_properties['invert_x_check'] = self.invert_x_check.isChecked()
+        layout_properties['x_axis_mode_combo'] = self.x_axis_mode_combo.currentText()
+        layout_properties['invert_y_check'] = self.invert_y_check.isChecked()
+        layout_properties['y_axis_mode_combo'] = self.y_axis_mode_combo.currentText()
+        layout_properties['orientation_combo'] = self.orientation_combo.currentText()
+        layout_properties['bar_mode_combo'] = self.bar_mode_combo.currentText()
+        layout_properties['hist_norm_combo'] = self.hist_norm_combo.currentText()
+        layout_properties['box_statistic_combo'] = self.box_statistic_combo.currentText()
+        layout_properties['outliers_combo'] = self.outliers_combo.currentText()
+        layout_properties['violinSideCombo'] = self.violinSideCombo.currentText()
+        layout_properties['showMeanCheck'] = self.showMeanCheck.isChecked()
+        layout_properties['cumulative_hist_check'] = self.cumulative_hist_check.isChecked()
+        layout_properties['invert_hist_check'] = self.invert_hist_check.isChecked()
+        layout_properties['bins_check'] = self.bins_check.isChecked()
+
         return PlotSettings(plot_type=self.ptype, properties=plot_properties, layout=layout_properties)
 
+    def set_settings(self):
+        """
+        Takes a PlotSettings object and fill the widgets with the settings
+        """
+
+        settings = self.get_settings()
+
+        # Set the plot properties
+        self.layer_combo.setLayer(settings.properties['layer_id'])
+        self.selected_feature_check.setChecked(settings.properties['features_selected'])
+        self.x_combo.setExpression(settings.properties['x_combo'])
+        self.y_combo.setExpression(settings.properties['y_combo'])
+        self.z_combo.setExpression(settings.properties['z_combo'])
+        self.in_color_combo.setColor(settings.properties['in_color_combo'])
```

Comment:
mental note: we'll need to convert colors to strings for storage. leave that one with me.